### ISSUE TITLE
Enable swagger on remote server when do_deploy_api()

### ISSUE THREAD

**Issue by rmnppt** [trestletech/plumber#208]
Hi, Could we please enable swagger by default when deploying an api via do_deploy_api().
As per the stackoverflow conversation here:
https://stackoverflow.com/questions/47203758/r-plumber-api-deployment-on-digital-ocean-with-swagger?noredirect=1#comment81364477_47203758

**Comment by Btibert3** [comment]
This would be fantastic to have.

**Comment by Btibert3** [comment]
After diving in, it looks like the swagger api isn't published due to the `interactive()` default value in` `run`  [found here](https://github.com/trestletech/plumber/blob/master/R/plumber.R#L190)

Is there a way to set this when running non-interactively without going down the path of building the API programmatically?

**Comment by Btibert3** [comment]
@rmnppt look at the string above.  Right now, `run` has a default value of `interactive()`.  I am hardly a programmer, so I wasnt sure the best way to test this, but by explicitly setting `swagger=true` within the Dockerfile, I was able to expose the swagger documentation on a remote Digital Ocean server.

https://hub.docker.com/r/btibert3/test-plumber-docker/~/dockerfile/

This is only meant to be a short run solution, but hopefully it highlights how to enable swagger in other ways beyond interactive mode.

**Comment by rmnppt** [comment]
@Btibert3 good idea, i'll do the same for now. thanks for looking into it.

**Comment by rmnppt** [comment]
here is my work around for now:

`droplet_execute(deepminer_api, {
  system("sed -i 's/pr$run(port=8000)/pr$run(port=8000, swagger = TRUE)/g' /etc/systemd/system/plumber-deepminer.service", intern = TRUE) 
  system("systemctl daemon-reload", intern = TRUE)
  system("service plumber-deepminer restart", intern = TRUE)
})`

**Comment by aleDsz** [comment]
What can I do to enable swagger on Shinyapps?

**Comment by trestletech** [comment]
I've got this supported in a branch. 

```
devtools::install_github("trestletech/plumber", ref="do-swagger")
```

You should now be able to run `do_deploy_api()` with `swagger=TRUE` to enable Swagger on the remote API. It's still opt-in, so you'll need to set that option if you want the swagger interface.

@aleDsz, ShinyApps.io doesn't currently support Plumber, but it's on their backlog.

**Comment by aleDsz** [comment]
@trestletech Actually, they support, but you need to use the `rsconnect` package to deploy it.

**Comment by trestletech** [comment]
Right, you can upload them but there are a few kinks we want to work out before they'd officially claim support.

**Comment by rmnppt** [comment]
cool, thank you!
